Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp

```diff
@@ -610,4 +627,114 @@ namespace AZ::Render
             m_fontDrawInterface->DrawScreenAlignedText3d(m_drawParams, joint->GetName());
         }
     }
+
+    void AtomActorDebugDraw::RenderNodeOrientations(EMotionFX::ActorInstance* actorInstance,
+        AzFramework::DebugDisplayRequests* debugDisplay, float scale)
+    {
+        // Get the actor and the transform data
+        const float unitScale =
+            1.0f / (float)MCore::Distance::ConvertValue(1.0f, MCore::Distance::UNITTYPE_METERS, EMotionFX::GetEMotionFX().GetUnitType());
+        const EMotionFX::Actor* actor = actorInstance->GetActor();
+        const EMotionFX::Skeleton* skeleton = actor->GetSkeleton();
+        const EMotionFX::TransformData* transformData = actorInstance->GetTransformData();
+        const EMotionFX::Pose* pose = transformData->GetCurrentPose();
+        const float constPreScale = scale * unitScale * 3.0f;
+
+        const size_t numEnabled = actorInstance->GetNumEnabledNodes();
+        for (size_t i = 0; i < numEnabled; ++i)
+        {
+            EMotionFX::Node* joint = skeleton->GetNode(actorInstance->GetEnabledNode(i));
+            const size_t jointIndex = joint->GetNodeIndex();
+
+            static const float axisBoneScale = 50.0f;
+            const float size = CalculateBoneScale(actorInstance, joint) * constPreScale * axisBoneScale;
+            AZ::Transform worldTM = pose->GetWorldSpaceTransform(jointIndex).ToAZTransform();
+            RenderLineAxis(debugDisplay, worldTM, size, false /*selected*/);
```

Comment:
What about early-out here for tiny joints? Maybe if the joint size is smaller than 1% of the character AABB, early out and skip rendering the axis?

Reply:
That's a hard line to find. Also it's possible that the user "just want to see that one tiny joint's axis".
In the future we could do some optimization, like only show those axis from Tiny joint when the view it's close (kinda like a mini LOD system within the viewport). For now i don't think it is necessary.